src/dmd/dinterpret.d

```diff
@@ -2776,14 +2777,38 @@ public:
                             return;
                         (*exps)[i] = ex;
                     }
+
                 }
                 sd.fill(e.loc, exps, false);
 
+                foreach (i, v; sd.fields)
+                {
+                    Expression exp = (*exps)[i];
+                    if (!exp)
+                        continue;
+                    Expression ex;
+                    if ((v.type.ty != exp.type.ty) && v.type.ty == Tsarray)
+                    {
+                        // Block assignment from inside struct literals
+                        auto tsa = cast(TypeSArray)v.type;
+                        auto len = cast(size_t)tsa.dim.toInteger();
+                        UnionExp ue = void;
+                        ex = createBlockDuplicatedArrayLiteral(&ue, exp.loc, v.type, exp, len);
+                        if (ex == ue.exp())
+                            ex = ue.copy();
+                    }
+                    if (ex && ex !is exp)
+                    {
+                        exps = copyArrayOnWrite(exps, e.arguments);
+                        (*exps)[i] = ex;
+                    }
+                }
+
                 auto se = ctfeEmplaceExp!StructLiteralExp(e.loc, sd, exps, e.newtype);
                 se.origin = se;
                 se.type = e.newtype;
                 se.ownedByCtfe = OwnedBy.ctfe;
-                result = interpret(pue, se, istate);
+                result = se;
```

Comment:
this doesn't look correct.
could skip resolving steps.

Reply:
nope. look at `visit(StructLiteralExp)` inside `interpret`. It returns immediately in all cases.